lush.nvim-scm-1.rockspec

```diff
@@ -0,0 +1,53 @@
+local _MODREV, _SPECREV = "scm", "-1"
+rockspec_format = "3.0"
+package = "lush.nvim"
+version = _MODREV .. _SPECREV
+
+description = {
+	summary = "Define Neovim themes as a DSL in lua, with real-time feedback.",
+	labels = { "neovim" },
+	detailed = [[
+    Lush is a colorscheme creation aid, written in Lua, for Neovim.
+   ]],
+	homepage = "https://github.com/rktjmp/lush.nvim",
+	license = "MIT/X11",
+}
+
+dependencies = {
+	"lua >= 5.1, < 5.4",
+}
+
+source = {
+	url = "http://github.com/rktjmp/lush.nvim/archive/v" .. _MODREV .. ".zip",
+	dir = "lush.nvim-" .. _MODREV,
+}
+
+if _MODREV == "scm" then
+	source = {
+		url = "git://github.com/rktjmp/lush.nvim",
+	}
+end
+
+build = {
+	type = "builtin",
+	modules = {
+		["lush"] = "lua/lush.lua",
+		["lush.hsl.convert"] = "lua/lush/hsl/convert.lua",
+		["lush.vivid.hsl.convert"] = "lua/lush/vivid/hsl/convert.lua",
+		["lush.vivid.hsl.type"] = "lua/lush/vivid/hsl/type.lua",
+		["lush.vivid.hsluv.convert"] = "lua/lush/vivid/hsluv/convert.lua",
+		["lush.vivid.hsluv.lib"] = "lua/lush/vivid/hsluv/lib.lua",
+		["lush.vivid.hsluv.type"] = "lua/lush/vivid/hsluv/type.lua",
+		["lush.vivid.rgb.convert"] = "lua/lush/vivid/rgb/convert.lua",
+		["lush.vivid.hsl_like"] = "lua/lush/vivid/hsl_like.lua",
+		["lush.compiler"] = "lua/lush/compiler.lua",
+		["lush.errors"] = "lua/lush/errors.lua",
+		["lush.hsl"] = "lua/lush/hsl.lua",
+		["lush.ify"] = "lua/lush/ify.lua",
+		["lush.math"] = "lua/lush/math.lua",
+		["lush.parser"] = "lua/lush/parser.lua",
```

Comment:
Not needed?

> To make writing rocks for projects with a common directory layout even easier, the "builtin" build type recursively copies all files and directories it finds in a folder called "./lua" (if it exists) to the same place as other Lua files. It's almost as if an entry had been added to modules for every file in lua, except that the files will only be copied - files ending in ".c" won't be compiled.

https://github.com/luarocks/luarocks/wiki/Creating-a-rock

Reply:
indeed ! I was actually going to complain about this to luarocks (I would prefer to have a glob("**/*.lua") like what bazel has rather than some magical include but that's nice.

I've tested the integration of this rockspec into nixpkgs and running the tests, I end up with:

```
Error → (tail call) @ -1
transforms run.lua returns lua code
./lua/shipwright/transform/lush/to_lua.lua:27: bad argument #3 to 'format' (string expected, got boolean)
```

Do you see something similar or it's just some version artifact ?

Reply:
using busted 2.0.0-1

Reply:
Semi-bug I guess:

```
~
λ lua
Lua 5.4.2  Copyright (C) 1994-2020 Lua.org, PUC-Rio
> string.format("%s", true)
true
> ^C

◷ 2s

~
λ lua
Lua 5.1.5  Copyright (C) 1994-2012 Lua.org, PUC-Rio
> string.format("%s", true)
stdin:1: bad argument #2 to 'format' (string expected, got boolean)
stack traceback:
	[C]: in function 'format'
	stdin:1: in main chunk
	[C]: ?
> ^C

◷ 3s
~
λ asdf shell lua-jit 2.1.0-beta3--3.4.0

~
λ luajit
LuaJIT 2.1.0-beta3 -- Copyright (C) 2005-2017 Mike Pall. http://luajit.org/
JIT: ON SSE2 SSE3 SSE4.1 BMI2 fold cse dce fwd dse narrow loop abc sink fuse
> string.format("%s", true)
> = string.format("%s", true)
true

◷ 5s
```

I probably need to add `tostring` wrappers here: :vomiting_face: 

https://github.com/rktjmp/lush.nvim/blob/6923ccef17c5276bc5d3cca907082e19ee165d55/lua/shipwright/transform/lush/to_lua.lua#L24-L27